### ISSUE TITLE
Tooltip exception: Bad state: Instance of '_OverlayEntryLocation' is already disposed

### ISSUE THREAD

**Issue by tgucio** [flutter/flutter#129025]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

Run the app with the sample code. Tap on the widgets to expand them a few times and close using the Done `IconButton`. After a few times the issue is seen.

### Expected results

No exceptions.

### Actual results

The following `StateError` is thrown:

```
Exception has occurred.
StateError (Bad state: Instance of '_OverlayEntryLocation' is already disposed. Stack trace: #0      _OverlayEntryLocation._dispose.<anonymous closure> (package:flutter/src/widgets/overlay.dart:1739:45)
#1      _OverlayEntryLocation._dispose (package:flutter/src/widgets/overlay.dart:1741:6)
#2      _OverlayPortalState.hide (package:flutter/src/widgets/overlay.dart:1603:21)
#3      OverlayPortalController.hide (package:flutter/src/widgets/overlay.dart:1351:13)
#4      TooltipState._handleStatusChanged (package:flutter/src/material/tooltip.dart:437:30)
#5      AnimationLocalStatusListenersMixin.notifyStatusListeners (package:flutter/src/animation/listener_helpers.dart:240:19)
#6      AnimationController._checkStatusChanged (package:flutter/src/animation/animation_controller.dart:815:7)
#7      AnimationController._tick (package:flutter/src/animation/animation_controller.dart:831:5)
#8      Ticker._tick (package:flutter/src/scheduler/ticker.dart:249:12)
#9      SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1297:15)
#10     SchedulerBinding.handleBeginFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:1149:11)
#11     _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:625:13)
#12     SchedulerBinding.handleBeginFrame (package:flutter/src/scheduler/binding.dart:1147:17)
#13     SchedulerBinding._handleBeginFrame (package:flutter/src/scheduler/binding.dart:1064:5)
#14     _invoke1 (dart:ui/hooks.dart:186:13)
#15     PlatformDispatcher._beginFrame (dart:ui/platform_dispatcher.dart:374:5)
#16     _beginFrame (dart:ui/hooks.dart:129:31)
```

### Code sample

<details><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {

  Future<void> _expandHeroWidget(String tag) {
    final Color barrierColor = Theme.of(context).colorScheme.onBackground.withOpacity(0.25);
    final HeroDialogRoute route = HeroDialogRoute(
      barrierColor: barrierColor,
      builder: (BuildContext context) {
        return Center(
          child: Padding(
            padding: const EdgeInsets.all(20.0),
            child: ConstrainedBox(
              constraints: const BoxConstraints(
                maxWidth: 600.0,
                maxHeight: 400.0,
              ),
              child: Hero(
                tag: tag,
                child: Material(
                  type: MaterialType.canvas,
                  color: Colors.amberAccent,
                  child: Row(
                    children: <Widget>[
                      IconButton(
                        onPressed: () {
                          final SnackBar snackBar = SnackBar(content: Text('SnackBar! $tag'));
                          ScaffoldMessenger.of(context).showSnackBar(snackBar);
                        },
                        icon: const Icon(Icons.abc),
                        tooltip: 'Show a snackbar',
                      ),
                      IconButton(
                        onPressed: () => Navigator.of(context).pop(),
                        icon: const Icon(Icons.check),
                        tooltip: 'Done',
                      ),
                    ],
                  ),
                ),
              ),
            ),
          ),
        );
      },
    );

    return Navigator.of(context).push(route);
  }

  Widget _buildHeroWidget(String tag) {
    return Hero(
      tag: tag,
      child: Container(
        color: Colors.amber,
        width: 300.0,
        height: 200.0,
        child: Material(
          type: MaterialType.transparency,
          child: InkResponse(
            onTap: () => _expandHeroWidget(tag),
            child: RichText(
              text: const TextSpan(text: 'tap me'),
              overflow: TextOverflow.fade,
            ),
          ),
        ),
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          mainAxisSize: MainAxisSize.min,
          children: <Widget>[ 
            _buildHeroWidget('widget1'),
            const SizedBox(height: 10.0),
            _buildHeroWidget('widget2'),
            const SizedBox(height: 10.0),
            _buildHeroWidget('widget3'),
          ],
        ),
      ),
    );
  }
}

class HeroDialogRoute<T> extends PageRoute<T> {
  HeroDialogRoute({this.barrierColor = const Color(0x80000000), required this.builder})
      : super();

  final WidgetBuilder builder;

  @override
  bool get opaque => false;

  @override
  bool get barrierDismissible => true;

  @override
  String? get barrierLabel => null;

  @override
  final Color barrierColor;

  @override
  Duration get transitionDuration => const Duration(milliseconds: 200);

  @override
  bool get maintainState => true;

  @override
  Widget buildTransitions(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget child) {
    return FadeTransition(
      opacity: CurvedAnimation(parent: animation, curve: Curves.easeOut),
      child: child,
    );
  }

  @override
  Widget buildPage(
    BuildContext context,
    Animation<double> animation,
    Animation<double> secondaryAnimation,
  ) {
    return builder(context);
  }
}
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
% flutter doctor -v
[✓] Flutter (Channel master, 3.12.0-4.0.pre.102, on macOS 13.4 22F66 darwin-arm64, locale en-ES)
    • Flutter version 3.12.0-4.0.pre.102 on channel master at /Users/tgucio/Library/Frameworks/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision dc4541fa05 (15 minutes ago), 2023-06-16 19:00:21 +0100
    • Engine revision fb5fed432e
    • Dart version 3.1.0 (build 3.1.0-213.0.dev)
    • DevTools version 2.24.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /Users/tgucio/Library/Android/sdk
    • Platform android-33, build-tools 33.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.79.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.66.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 13.4 22F66 darwin-arm64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 114.0.5735.133

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by tgucio** [comment]
Possibly this started happening after https://github.com/flutter/flutter/pull/127728
/cc @LongCatIsLooong

**Comment by huycozy** [comment]
Hi @tgucio, it would be great if there is a reproduction sample for this.

**Comment by tgucio** [comment]
Looks like I've been able to create a mini repro:

```
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {

  Future<void> _expandHeroWidget(String tag) {
    final Color barrierColor = Theme.of(context).colorScheme.onBackground.withOpacity(0.25);
    final HeroDialogRoute route = HeroDialogRoute(
      barrierColor: barrierColor,
      builder: (BuildContext context) {
        return Center(
          child: Padding(
            padding: const EdgeInsets.all(20.0),
            child: ConstrainedBox(
              constraints: const BoxConstraints(
                maxWidth: 600.0,
                maxHeight: 400.0,
              ),
              child: Hero(
                tag: tag,
                child: Material(
                  type: MaterialType.canvas,
                  color: Colors.amberAccent,
                  child: Row(
                    children: <Widget>[
                      IconButton(
                        onPressed: () {
                          final SnackBar snackBar = SnackBar(content: Text('SnackBar! $tag'));
                          ScaffoldMessenger.of(context).showSnackBar(snackBar);
                        },
                        icon: const Icon(Icons.abc),
                        tooltip: 'Show a snackbar',
                      ),
                      IconButton(
                        onPressed: () => Navigator.of(context).pop(),
                        icon: const Icon(Icons.check),
                        tooltip: 'Done',
                      ),
                    ],
                  ),
                ),
              ),
            ),
          ),
        );
      },
    );

    return Navigator.of(context).push(route);
  }

  Widget _buildHeroWidget(String tag) {
    return Hero(
      tag: tag,
      child: Container(
        color: Colors.amber,
        width: 300.0,
        height: 200.0,
        child: Material(
          type: MaterialType.transparency,
          child: InkResponse(
            onTap: () => _expandHeroWidget(tag),
            child: RichText(
              text: const TextSpan(text: 'tap me'),
              overflow: TextOverflow.fade,
            ),
          ),
        ),
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          mainAxisSize: MainAxisSize.min,
          children: <Widget>[ 
            _buildHeroWidget('widget1'),
            const SizedBox(height: 10.0),
            _buildHeroWidget('widget2'),
            const SizedBox(height: 10.0),
            _buildHeroWidget('widget3'),
          ],
        ),
      ),
    );
  }
}

class HeroDialogRoute<T> extends PageRoute<T> {
  HeroDialogRoute({this.barrierColor = const Color(0x80000000), required this.builder})
      : super();

  final WidgetBuilder builder;

  @override
  bool get opaque => false;

  @override
  bool get barrierDismissible => true;

  @override
  String? get barrierLabel => null;

  @override
  final Color barrierColor;

  @override
  Duration get transitionDuration => const Duration(milliseconds: 200);

  @override
  bool get maintainState => true;

  @override
  Widget buildTransitions(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget child) {
    return FadeTransition(
      opacity: CurvedAnimation(parent: animation, curve: Curves.easeOut),
      child: child,
    );
  }

  @override
  Widget buildPage(
    BuildContext context,
    Animation<double> animation,
    Animation<double> secondaryAnimation,
  ) {
    return builder(context);
  }
}
```

**Comment by huycozy** [comment]
Thanks for your update. Reproduced this issue on the latest master channel whilst it doesn't appear on stable. 

<details>
<summary>Complete output log</summary>

```console
🔥  To hot restart changes while running, press "r" or "R".
For a more detailed help message, press "h". To quit, press "q".

A Dart VM Service on Chrome is available at: http://127.0.0.1:50310/FfoQYq4RAVc=
The platformViewRegistry getter is deprecated and will be removed in a future release. Please import it from `dart:ui_web` instead.
The Flutter DevTools debugger and profiler on Chrome is available at: http://127.0.0.1:9100?uri=http://127.0.0.1:50310/FfoQYq4RAVc=
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following StateError was thrown building Hero(tag: widget1, dependencies: [MediaQuery], state:
_HeroState#bded0):
Bad state: Instance of '_OverlayEntryLocation' is already disposed. Stack trace:
dart-sdk/lib/_internal/js_dev_runtime/patch/core_patch.dart 941:28                get current
packages/flutter/src/widgets/overlay.dart 1739:45                                 <fn>
packages/flutter/src/widgets/overlay.dart 1740:14                                 [_dispose]
packages/flutter/src/widgets/overlay.dart 1603:21                                 hide
packages/flutter/src/widgets/overlay.dart 1351:12                                 hide
packages/flutter/src/material/tooltip.dart 437:30
[_handleStatusChanged]
packages/flutter/src/animation/listener_helpers.dart 240:19
notifyStatusListeners
packages/flutter/src/animation/animation_controller.dart 815:7
[_checkStatusChanged]
packages/flutter/src/animation/animation_controller.dart 831:5                    [_tick]
packages/flutter/src/scheduler/ticker.dart 249:12                                 [_tick]
packages/flutter/src/scheduler/binding.dart 1297:15
[_invokeFrameCallback]
packages/flutter/src/scheduler/binding.dart 1149:11                               <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/linked_hash_map.dart 21:13          forEach
packages/flutter/src/scheduler/binding.dart 1147:16                               handleBeginFrame
packages/flutter/src/scheduler/binding.dart 1064:5
[_handleBeginFrame]
lib/_engine/engine/platform_dispatcher.dart 1319:13                               invoke1
lib/_engine/engine/platform_dispatcher.dart 257:5                                 invokeOnBeginFrame
lib/_engine/engine/initialization.dart 176:45                                     <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 574:37  _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 579:39  dcall

The relevant error-causing widget was:
  Hero
  Hero:file:///Users/huynq/Documents/NEVERCODE.IO/RepoduceProjects/reproduce_issues_2/lib/tooltip_exp_129025.dart:47:22

When the exception was thrown, this was the stack:
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 294:49      throw_
packages/flutter/src/widgets/overlay.dart 1731:5                                  [_debugNotDisposed]
packages/flutter/src/widgets/overlay.dart 1722:12                                 [_deactivate]
packages/flutter/src/widgets/overlay.dart 1884:22                                 deactivate
packages/flutter/src/widgets/framework.dart 1971:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 6421:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 6421:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 6421:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 6421:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 6421:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 6421:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 6421:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 6421:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 6421:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 6421:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 6421:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 6421:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 6421:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 6533:16                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 6421:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 6421:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 5350:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 6421:14                               visitChildren
packages/flutter/src/widgets/framework.dart 1973:12                               _deactivateRecursively
packages/flutter/src/widgets/framework.dart 1985:7                                add
packages/flutter/src/widgets/framework.dart 4225:5                                deactivateChild
packages/flutter/src/widgets/framework.dart 3631:9                                updateChild
packages/flutter/src/widgets/framework.dart 6442:14                               update
packages/flutter/src/widgets/framework.dart 3686:14                               updateChild
packages/flutter/src/widgets/framework.dart 5323:16                               performRebuild
packages/flutter/src/widgets/framework.dart 5463:11                               performRebuild
packages/flutter/src/widgets/framework.dart 5017:7                                rebuild
packages/flutter/src/widgets/framework.dart 2780:18                               buildScope
packages/flutter/src/widgets/binding.dart 916:9                                   drawFrame
packages/flutter/src/rendering/binding.dart 361:5                                 [_handlePersistentFrameCallback]
packages/flutter/src/scheduler/binding.dart 1297:15                               [_invokeFrameCallback]
packages/flutter/src/scheduler/binding.dart 1227:9                                handleDrawFrame
packages/flutter/src/scheduler/binding.dart 1085:5                                [_handleDrawFrame]
lib/_engine/engine/platform_dispatcher.dart 1304:13                               invoke
lib/_engine/engine/platform_dispatcher.dart 278:5                                 invokeOnDrawFrame
lib/_engine/engine/initialization.dart 185:45                                     <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 574:37  _checkAndCall
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 579:39  dcall

════════════════════════════════════════════════════════════════════════════════════════════════════

```
</details>

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.10.5, on macOS 13.0.1 22A400 darwin-x64, locale en-VN)
    • Flutter version 3.10.5 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 796c8ef792 (29 hours ago), 2023-06-13 15:51:02 -0700
    • Engine revision 45f6e00911
    • Dart version 3.0.5
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 32.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.79.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.66.0

[✓] Connected device (5 available)
    • Pixel 3a (mobile)      • 964AY0WL20                               • android-arm64  • Android 12 (API 32)
    • iPhone (mobile)        • d9a94afe2b649fef56ba0bfeb052f0f2a7dae95e • ios            • iOS 15.7.2 19H218
    • iPhone 14 Pro (mobile) • 98082851-A0C4-4E39-A310-21C48B10A364     • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • macOS (desktop)        • macos                                    • darwin-x64     • macOS 13.0.1 22A400 darwin-x64
    • Chrome (web)           • chrome                                   • web-javascript • Google Chrome 114.0.5735.133

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

```bash
[!] Flutter (Channel master, 3.12.0-4.0.pre.121, on macOS 13.0.1 22A400 darwin-x64, locale en-VN)
    • Flutter version 3.12.0-4.0.pre.121 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not
      inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding
      /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside
      your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding
      /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5182697c0d (2 hours ago), 2023-06-19 21:37:33 -0400
    • Engine revision a91bb3f566
    • Dart version 3.1.0 (build 3.1.0-230.0.dev)
    • DevTools version 2.24.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly
      to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 32.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.79.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.66.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 13.0.1 22A400 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 114.0.5735.133

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```
</details>
